### PR TITLE
Issue 225 make the container grid class configurable from the schema

### PR DESCRIPTION
**Pull request by samottitusgt**

### CODE REVIEW

File: xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java [hunk 1]
```diff
@@ -300,7 +301,7 @@ public JSONObject generateNgDynamicFormsCls() {
         fieldElement.put("label", "ui-widget");
 
         JSONObject fieldGrid = new JSONObject();
-        fieldGrid.put("container", "ui-g");
+        fieldGrid.put("container", ngcontainer!=null? ngcontainer : "ui-g");
```
Comment (taurus227) [inline xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: use `StringUtils.isNotBlank(ngcontainer)`

File: xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java [hunk 2]
```diff
@@ -374,6 +375,8 @@ public JSONObject getCommonFieldsNgDynamicForms() {
         //This can overwrite values set earlier, for example 'type' can be changed from INPUT to RATING
         readAppInfoDynamicForms(model, field);
 
+        field.put("cls", generateNgDynamicFormsCls());
```
Comment (taurus227) [inline xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: Changing the order might have significant side-effects. Why is this needed?

File: xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java [hunk 3]
```diff
@@ -327,6 +327,9 @@ protected void setAppInfoDynamicFormsJsonValue(AnyNode node, JSONObject json) {
             else if (name.equals("errmsg")) {
                 errmsg = value;
             }
+            else if (name.equals("container")) {
+                ngcontainer = value;
+            }
```
Comment (taurus227) [inline xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: Add unit test both to dsl and to outcome builder.

File: xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java (hunk 2 above)
Comment (samottitusgt) [reply xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: I test it and looked into the order, it does not have any side-effects. It needs to run first readAppInfoDynamicForms(model, field) to set the value for ngcontainer.

File: xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java (hunk 1 above)
Comment (samottitusgt) [reply xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: will apply this

File: xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java (hunk 3 above)
Comment (samottitusgt) [reply xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: will add this to unit test.

Comment (dokithonon-icube) [reply xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/StringField.java]: Can you provide the wiki link so that I know how to confîgure it ? thanks
